lib/semantics/resolve-names.cc

```diff
@@ -3796,8 +3800,15 @@ bool DeclarationVisitor::PassesLocalityChecks(
         "Assumed size array '%s' not allowed in a locality-spec"_err_en_US);
     return false;
   }
-  // TODO: Check to see if the name can appear in a variable definition context
-  return true;
+  if (std::optional<MessageFixedText> msg{
+          WhyNotModifiable(symbol, currScope())}) {
+    SayWithReason(name, symbol,
+        "'%s' must be able to appear in a variable definition context to "
+        "appear in a locality-spec"_err_en_US,
```

Comment:
This message is precise, but maybe too technical.
What do you think of this:
"'%s' may not appear in a locality-spec because it is not a definable variable"

Reply:
Thanks.  I'll make this change.

Reply:
Actually, I think I'll use this message here:
"'%s' may not appear in a locality-spec because it is not definable"

The potential problem is that we augment this message with another message that gives the reason that it's not definable.  Sometimes, the reason states that it's not a variable.  If both of these messages mention not being a variable it reads awkwardly.